wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -39,6 +39,48 @@ private[wallet] trait UtxoHandling extends WalletLogger {
     spendingInfoDAO.findAllUnspentForAccount(hdAccount)
   }
 
+  protected def updateUtxoConfirmedState(
+      txo: SpendingInfoDb,
+      blockHash: DoubleSha256DigestBE): Future[SpendingInfoDb] = {
+    for {
+      confsOpt <- chainQueryApi.getNumberOfConfirmations(blockHash)
+      stateChange <- {
+        confsOpt match {
+          case None => Future.successful(txo)
+          case Some(confs) =>
+            txo.state match {
+              case TxoState.PendingConfirmationsReceived |
+                  TxoState.DoesNotExist =>
+                if (confs >= walletConfig.requiredConfirmations) {
+                  spendingInfoDAO.update(
+                    txo.copyWithState(TxoState.ConfirmedReceived))
+                } else {
+                  Future.successful(
+                    txo.copyWithState(TxoState.PendingConfirmationsReceived))
+                }
+              case TxoState.PendingConfirmationsSpent =>
+                if (confs >= walletConfig.requiredConfirmations) {
+                  spendingInfoDAO.update(
+                    txo.copyWithState(TxoState.ConfirmedSpent))
+                } else {
+                  Future.successful(txo)
+                }
+              case TxoState.Reserved =>
+                // We should keep the utxo as reserved so it is not used in
+                // a future transaction that it should not be in
+                Future.successful(txo)
+              case TxoState.ConfirmedReceived | TxoState.ConfirmedSpent =>
+                val errMsg =
+                  s"TXO ${txo.txid} was marked as confirmed but did not have a block hash, the issue has been fixed"
```

Comment:
how is this fixed? I don't see it getting updated with the hash? Also isn't this technically the wrong hash? Is this why we aren't calling `update`? Couldn't this be a reorg?